### PR TITLE
Refactor core errors

### LINKED ISSUE #191: Better errors formatting in config API response
For now, we have an endpoint `/api/v1/config` that returns something like this:

```json
{
  "version": 1,
  "tiles": [
    {"type": "EMPTYYYY"}
  ],
  "errors": ["Unsupported version", "Columns field is missing", "Unknown tile type EMPTYYYY"]
}
```

That way, we mix `config` itself and errors added by the *Core*'s Verify method.

For better errors management, we need to:
- split config and errors
- add a unique ID also used as a translation key
- get the raw value which causes the error (configUrl, specific tile's JSON, ...)

With that in mind, we can go to something more like that:

```json
{
  "config": {
    "version": 1,
    "tiles": [
      {"type": "EMPTYYYY"}
    ]
  },
  "errors": [
    {
      "id": "ERROR_UNSUPPORTED_VERSION",
      "message": "Unsupported version",
      "data": {
        "configExtract": "{\"version\": 1}",
        "expected": ">= 4"
      }
    },
    {
      "id": "ERROR_MISSING_REQUIRED_FIELD",
      "message": "Columns field is missing",
      "data": {
        "fieldName": "columns",
        "expected": ">= 1"
      }
    },
    {
      "id": "ERROR_UNKNOWN_TILE_TYPE",
      "message": "Unknown tile type EMPTYYYY",
      "data": {
        "configExtract": "{\"type\": \"EMPTYYYY\"}",
        "expected": "EMPTY, GROUP, PING, PORT, [...]"
      }
    }
  ]
}
```

### PR DESCRIPTION
**Pull request by Alex-D**
fix #191 #199 #200 #201

### CODE REVIEW

File: monitorable/config/models/config.go [hunk 1]
```diff
@@ -24,15 +22,39 @@ type (
...
+	}
+
+	ConfigErrorData struct {
+		ConfigExtract string `json:"configExtract,omitempty"`
+		Value         string `json:"value,omitempty"`
+		FieldName     string `json:"fieldName,omitempty"`
+		Expected      string `json:"expected,omitempty"`
+	}
+
+	ConfigBag struct {
```
Comment (jsdidierlaurent) [inline monitorable/config/models/config.go]: Usualy, i sort structures by hierarchy. Put this on top.

File: monitorable/config/models/configversion.go [hunk 2]
```diff
@@ -39,52 +40,38 @@ func (v *ConfigVersion) UnmarshalJSON(data []byte) error {
...
 }
 
-func parseVersion(version string) (*ConfigVersion, error) {
+func parseVersion(version string) *ConfigVersion {
+	version = strings.ReplaceAll(fmt.Sprintf(`"%s"`, version), `""`, `"`)
 	v := &ConfigVersion{}
 	err := v.UnmarshalJSON([]byte(version))
 	if err != nil {
-		return nil, err
+		panic(fmt.Sprintf(`Invalide version format used in configversion comparaison function "%s", %v`, version, err))
```
Comment (Alex-D) [inline monitorable/config/models/configversion.go]: Invalid, without e

File: monitorable/config/models/configversion.go [hunk 3]
```diff
@@ -39,52 +40,38 @@ func (v *ConfigVersion) UnmarshalJSON(data []byte) error {
 	return nil
 }
 
-func (v *ConfigVersion) MustEqualTo(v2Str string) (bool, error) {
-	v2, err := parseVersion(v2Str)
-	if err != nil {
-		return false, err
-	}
-	return v.major == v2.major && v.minor == v2.minor, nil
+func (v *ConfigVersion) EqualTo(v2Str string) bool {
```
Comment (Alex-D) [inline monitorable/config/models/configversion.go]: Prefix with `Is` => `IsEqualTo`, to say the function return a bool

File: monitorable/config/models/configversion.go [hunk 4]
```diff
@@ -39,52 +40,38 @@ func (v *ConfigVersion) UnmarshalJSON(data []byte) error {
...
+	return v.major > v2.major || (v.major == v2.major && v.minor > v2.minor)
 }
 
-func (v *ConfigVersion) MustLessThan(v2Str string) (bool, error) {
-	v2, err := parseVersion(v2Str)
-	if err != nil {
-		return false, err
-	}
-	return v.major < v2.major || (v.major == v2.major && v.minor < v2.minor), nil
+func (v *ConfigVersion) LessThan(v2Str string) bool {
```
Comment (Alex-D) [inline monitorable/config/models/configversion.go]: `IsLowerThan` is more consistent with `IsGreaterThan` I think

File: monitorable/config/models/configversion.go [hunk 5]
```diff
@@ -39,52 +40,38 @@ func (v *ConfigVersion) UnmarshalJSON(data []byte) error {
...
+	return v.major < v2.major || (v.major == v2.major && v.minor < v2.minor)
 }
 
-func (v *ConfigVersion) MustGreaterThanOrEqualTo(v2Str string) (bool, error) {
-	v2, err := parseVersion(v2Str)
-	if err != nil {
-		return false, err
-	}
-	return v.major > v2.major || (v.major == v2.major && v.minor > v2.minor) || (v.major == v2.major && v.minor == v2.minor), nil
+func (v *ConfigVersion) GreaterThanOrEqualTo(v2Str string) bool {
```
Comment (Alex-D) [inline monitorable/config/models/configversion.go]: `IsGreaterOrEqualThan`

**Pull request merged**